invoices/update.go

```diff
@@ -206,15 +229,91 @@ func updateMpp(ctx *invoiceUpdateCtx,
 		return &update, ctx.acceptRes(resultAccepted), nil
 	}
 
+	var (
+		htlcPreimages map[channeldb.CircuitKey]lntypes.Preimage
+		htlcPreimage  lntypes.Preimage
+	)
+	if ctx.amp != nil {
+		// Create a slice containing all the child descriptors to be
+		// used for reconstruction. This should include all HTLCs
+		// currently in the HTLC set, plus the incoming HTLC.
+		childDescs := make([]amp.ChildDesc, 0, 1+len(htlcSet))
+
+		// Add the new HTLC's child descriptor at index 0.
+		childDescs = append(childDescs, amp.ChildDesc{
```

Comment:
Is the ordering here critical?

Reply:
the order doesn't matter, only matters that the child descriptors are mapped properly to their HTLCs to check the payment hashes